apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -6,15 +6,96 @@ import {
   CommandLineStringParameter,
   ICommandLineActionOptions
 } from '@rushstack/ts-command-line';
+import { SyncHook, AsyncParallelHook, AsyncSeriesHook } from 'tapable';
+import { performance } from 'perf_hooks';
 
 import { HeftActionBase, IHeftActionBaseOptions, ActionHooksBase, IActionDataBase } from './HeftActionBase';
+import { CleanAction } from './CleanAction';
 
-export interface IBuildActionOptions extends IHeftActionBaseOptions {}
+export interface IBuildActionOptions extends IHeftActionBaseOptions {
+  cleanAction: CleanAction;
+}
+
+/**
+ * @public
+ */
+export class BuildPhaseHooksBase {
+  public readonly run: AsyncParallelHook = new AsyncParallelHook();
+}
+
+/**
+ * @public
+ */
+export interface IBuildPhase<TBuildPhaseHooks extends BuildPhaseHooksBase = BuildPhaseHooksBase> {
+  hooks: TBuildPhaseHooks;
+}
+
+/**
+ * @public
+ */
+export interface ISharedCopyStaticAssetsConfiguration {
+  /**
+   * File extensions that should be copied from the src folder to the destination folder(s)
+   */
+  fileExtensions?: string[];
+
+  /**
+   * Paths or globs that should be explicitly excluded. This takes precedence over paths listed in \"include\".
+   */
+  exclude?: string[];
+
+  /**
+   * Paths or globs that should be explicitly included.
+   */
+  include?: string[];
+}
+
+/**
+ * @public
+ */
+export interface ICopyStaticAssetsConfiguration extends ISharedCopyStaticAssetsConfiguration {
+  /**
+   * The folder from which assets should be copied
+   */
+  sourceFolderName: string | undefined;
+
+  /**
+   * The folder(s) to which assets should be copied
+   */
+  destinationFolders: string[];
```

Comment:
Why is the source a `FolderName` whereas the destination is a `Folder`?

To me, "folder name" means `something-without-slashes` whereas "folder" means a folder path.

Reply:
I'll change `destinationFolders` to `destinationFolderNames`

Reply:
It also might be useful for the doc comment to give an example of each thing.